include/orizzonte/utility/nothing.hpp

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include "./fwd.hpp"
+#include <tuple>
+#include <type_traits>
+
+namespace orizzonte::utility
+{
+    struct nothing
+    {
+    };
+
+    inline constexpr nothing nothing_v{};
+
+    template <typename T>
+    using void_to_nothing_t = std::conditional_t<std::is_void_v<T>, nothing, T>;
```

Comment:
This is actually a misnomer, right? It's actually "void_to_nothing_or_same", which is a terrible name, but you get the point.